db/db_test2.cc

```diff
@@ -985,8 +992,12 @@ TEST_F(DBTest2, WalFilterTestWithChangeBatchExtraKeys) {
    WalProcessingOption LogRecord(const WriteBatch& batch, WriteBatch* new_batch,
                                  bool* batch_changed) const override {
      *new_batch = batch;
-     new_batch->Put("key_extra", "value_extra");
-     *batch_changed = true;
+     Status s = new_batch->Put("key_extra", "value_extra");
+     if (s.ok()) {
+       *batch_changed = true;
+     } else {
+       assert(false);
```

Comment:
or FAIL?

Reply:
`FAIL` can't be used here which is why I went with `assert(false)`:

```
db/db_test2.cc:999:8: error: cannot initialize return object of type 'rocksdb::WalFilter::WalProcessingOption' with an rvalue of type 'void'
       FAIL();
       ^~~~~~
```